samples/hello_fpga.c

```diff
@@ -106,6 +113,132 @@ void print_err(const char *s, fpga_result res)
 	fprintf(stderr, "Error %s: %s\n", s, fpgaErrStr(res));
 }
 
+/*
+ * Global configuration of bus, set during parse_args()
+ * */
+struct config{
+	struct target {
+		int bus;
+		int open_flags;
+	} target;
+}
+
+config = {
+	.target = {
+		.bus = -1,
+		.open_flags = 0
+	}
+};
+
+#define GETOPT_STRING "B:s"
+fpga_result parse_args(int argc, char *argv[])
+{
+	struct option longopts[] = {
+		{"bus",		required_argument, NULL, 'B'},
+		{"shared access", no_argument,     NULL, 's'}
+	};
+	
+	int getopt_ret;
+	int option_index;
+	char *endptr = NULL;
+
+	while (-1 != (getopt_ret = getopt_long(argc, argv, GETOPT_STRING,
+						longopts, &option_index))) {
+		const char *tmp_optarg = optarg;
+		/* Checks to see if optarg is null and if not it goes to value of optarg */
+		if ((optarg) && ('=' == *tmp_optarg)){
+			++tmp_optarg;
+		}
+
+		switch (getopt_ret){
+		case 'B': /* bus */
+			if (NULL == tmp_optarg)
+				break;
```

Comment:
This is an error condition, as there is a required argument for -B.  Should print an error and exit instead of break and continue.